components/src/lib/components/ModalHeader.tsx

```diff
@@ -0,0 +1,49 @@
+import styled from "styled-components";
+import xButton from "./asset/xButton.svg";
+
+export interface Props {
+  title: string;
+  hasXButton?: boolean;
+  setModalClose: (e: React.MouseEvent) => void;
+  xButtonContent?: string;
+}
+const ModalHeader = ({
+  title,
+  hasXButton = true,
+  setModalClose,
+  xButtonContent,
+}: Props) => {
+  return (
+    <HeaderContainer>
+      {title}
+      <Button onClick={(e) => setModalClose(e)}>
+        {hasXButton && <img src={xButtonContent || xButton} />}
```

Comment:
xButton에 유저가 원하는 이미지를 넣게 확장성을 준 부분이 디테일하네요! 👍 
`xButtonContent`에 urlpath가 들어갈 것 같은데, 그럼 변수명에 나타내 주는 것도 좋을 것 같아요!
```suggestion
        {hasXButton && <img src={xButtonContentUrl || xButton} />}
```

Reply:
그런데 이 부분에, hasXButton이 false인 경우에, xButton의 이미지만 사라질 뿐, Button이 남아있을 수 있는 것 같아요.
이미지만 없고, 해당 영역의 버튼을 눌러서 닫기를 할 수 있는 현상이 발생하고 있어요!

![image](https://github.com/dle234/react-modules/assets/85233397/87564847-2828-448f-ae17-547999ba3e03)